hd/hd.c

```diff
@@ -0,0 +1,91 @@
+#include <stdio.h>
+#include <ctype.h>
+
+int t(int s, int c);
+
+int main()
+{
+  int s = 0;
+  int c;
+  while ((c = getchar()) != EOF) {
+    s = t(s, c);
+    if (s < 0) return s;
+    if (s == 5) {
+      printf("# ");
+    } else if (s == 9) {
+      printf(" #");
+    }
+    printf("%c", c);
+  }
+  return 0;
+}
+
+int t(int s, int c) {
```

Comment:
![image](https://github.com/sdbx/Utils/assets/83723320/2dae17f1-2b03-4abe-8689-50de04c8343c)
It would be nice if you could provide some explanations about this function UwU!!

Reply:
I rename it to a descriptive name `transition`.  19bb355fe23d854e5ddcf15aacb3b746e3ae22ab